### DISCUSSION TITLE
Set the MiddlePane to precise measurement in PX

### DISCUSSION THREAD

**Discussion by paulm17** [bvaughn/react-resizable-panels#395] [Q&A] [answered]
I have a 3 panel display.  

I'm wanting to set the middle panel to a precise pixel width and then have the 2 sides adjust to the remaining width.

I have an example here: [...]

I have tried multiple ways to attack this.  Always falling short.

I'm just not able to have the middle panel the same as the width of the intending device.  Always under or over.

Am I missing something fundamental here, or it's due to the library being configured for percentage sizing?

Thanks.

**Comment by bvaughn** [comment]
[Pixel based constraints are not supported.](https://github.com/bvaughn/react-resizable-panels?tab=readme-ov-file#can-panel-sizes-be-specified-in-pixels)

**Reply by paulm17** [reply]
I realise that and I was converting the values to percentage.

Ok, looks like I need to work out the difference between 0.0.63 and 2.1.1 to create a fork that support pixel throughout the library.

I noticed that 0.0.63 is a mixture between both.

**Reply by bvaughn** [reply]
Good luck!

**Reply by paulm17** [reply]
Thanks.  I'm hoping I can do it due to only requiring a fixed 3 panel vertical layout and removing a ton of options, to keep the edge cases minimal.  But we'll see!

**Reply by bvaughn** [reply]
I wonder if you might not have an easier time using a regular `<div>` in between two `<Panel>` components?

```jsx
<PanelGroup>
  <Panel />
  <div />
  <Panel />
</PanelGroup>
```

Just an idea 🤷🏼

**Reply by paulm17** [reply]
🤔 

I will try that first and see how I get on.  Thanks.

**Comment by bvaughn** [comment, marked as answer]
FYI pixel based constraints are supported in version 4 of this library
